homeassistant/components/mqtt/mixins.py

```diff
@@ -1274,7 +1276,7 @@ def _message_callback(
             _LOGGER.warning(exc)
             return
 
-        if self._attrs_have_changed(attrs_snapshot):
+        if attributes is not None and self._attrs_have_changed(attrs_snapshot):
             mqtt_data.state_write_requests.write_state_request(self)
```

Comment:
I was going to suggest to always write if its `None` but it looks like camera doesn't so I think this pattern is the way to go even if we end up with some duplicate code